### ISSUE TITLE
rate_num_examples Doesn't work with new tasks

### ISSUE THREAD

**Issue by agemagician** [google-research/text-to-text-transfer-transformer#41]
Hi,

I am training new mixture tasks (new tasks from tsv files), but it only works with "default_rate=1.0".
When I try to use "default_rate=rate_num_examples", it doesn't work because it can't find the tasks in the cache.

In issue #15 you mentioned the cache is only supported in your internal infrastructure.

So, how we can use the rate_num_examples in new mixture tasks ?

**Comment by craffel** [comment]
This is a known issue that we are working on. For now, you can hardcode the number of examples in each task when constructing the mixture.

**Comment by agemagician** [comment]
@craffel Thanks for the reply.

For now, I set it manually to:

t5.data.MixtureRegistry.add(
    "mi task",
    [["task1",0.34], ["task2",0.34],["task3",0.21],["task3",0.11]],
)

Hopefully, you fix it soon.

**Comment by craffel** [comment]
Thanks, yes, it is at the top of our to-do list!

**Comment by agemagician** [comment]
@craffel Is the rate is correct as in the above example or there is another way to do it ?

I have checked there is no number of examples for the mixture task, only the rate along each task.

I have checked the code and the task sampling is generated using "tf.data.experimental.sample_from_datasets" which uses "math_ops.log".

I am not sure should we pass the actual number of examples for each task or the rate of each task which is sum of 1 ?

using rates leads to :
[0.34, 0.34, 0.21, 0.11] -> [-1.0788096 -1.0788096 -1.5606477 -2.207275 ]
using actual number of samples :
[10791.0, 10791.0, 6621.0, 3744.0] -> [9.286468 9.286468 8.798001 8.22791 ]

each one of them leads to totally different results using the multi-task training.

**Comment by craffel** [comment]
It uses a log to convert the weights to logits before feeding them into a multinomial distribution for sampling. This is not a detail you need to worry about. It will sample proportionally according to the weights as you'd expect.

**Comment by roper-v** [comment]
test

**Comment by roper-v** [comment]
test
